tfjs-backend-webgl/src/im2col_packed_gpu.ts

```diff
@@ -17,38 +17,39 @@
 
 import {backend_util} from '@tensorflow/tfjs-core';
 import {getGlslDifferences} from './glsl_version';
-import {GPGPUProgram} from './gpgpu_math';
+import {GPGPUProgram, useShapeUniforms} from './gpgpu_math';
+import {UniformType} from './shader_compiler';
 
 export class Im2ColPackedProgram implements GPGPUProgram {
   variableNames = ['A'];
   packedInputs = true;
   packedOutput = true;
   outputShape: number[];
   userCode: string;
-
-  constructor(
-      outputShape: number[], inputShape: number[],
-      convInfo: backend_util.Conv2DInfo) {
+  enableShapeUniforms: boolean;
+  customUniforms = [
+    {name: 'inputShape', type: 'ivec3' as UniformType},
+    {name: 'pad', type: 'ivec2' as UniformType},
+    {name: 'stride', type: 'ivec2' as UniformType},
+    {name: 'dilation', type: 'ivec2' as UniformType},
+    {name: 'filterWidth', type: 'int' as UniformType},
+    {name: 'inChannels', type: 'int' as UniformType},
+    {name: 'itemsPerBlockRow', type: 'int' as UniformType},
+    {name: 'outWidth', type: 'int' as UniformType},
```

Comment:
Combine the last four `int` items into a `ivec4` uniform maybe save the uploading time a bit. What's your suggestion?